File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    just_lost = set(lost) - set(reserve)
    just_reserve = set(reserve) - set(lost)
    
    for i in just_reserve:
        if i-1 in just_lost:
            just_lost.remove(i-1)
            
        elif i+1 in just_lost:
            just_lost.remove(i+1)
    
    answer = n - len(just_lost)
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    let answer = n - lost.length;
    lost.sort((a, b) => a-b);
    const visit = Array.from({length: n+1}, ()=>(0));
    for (let i=0; i<n; i++) {
        const tmp = lost[i]
        if (reserve.includes(lost[i])) {
            visit[i] = 1;
            answer += 1;
        } else{
            if (reserve.includes(tmp-1) && !lost.includes(tmp-1) && visit[tmp-1] === 0) {
                visit[tmp-1] = 1;
                answer += 1;
            } else if (reserve.includes(tmp+1) && !lost.includes(tmp+1) && visit[tmp+1] === 0) {
                visit[tmp+1] = 1;
                answer += 1;
            }
        }
    }
    return answer;
}